tests/Doctrine/Tests/DBAL/Functional/Types/DecimalTest.php

```diff
@@ -32,4 +32,27 @@ public function testInsertAndRetrieveDecimal(): void
 
         self::assertSame('13.37', $value);
     }
+
+    public function testInsertAndRetrieveDecimalInteger(): void
```

Comment:
Please convert this to a data provider. Otherwise, it duplicates most of the test logic.